Looks like we are missing a dependency on ros-rmw:
```
/opt/ros/crystal/include/rcl/context.h:23:10: fatal error: rmw/init.h: No such file or directory
 #include "rmw/init.h"
          ^~~~~~~~~~~~
```

--- a/gazebo_ros/include/gazebo_ros/node.hpp
+++ b/gazebo_ros/include/gazebo_ros/node.hpp
@@ -136,22 +136,12 @@ Node::SharedPtr Node::CreateWithArgs(Args && ... args)
   // TODO(chapulina): use rclcpp::is_initialized() once that's available, see
   // https://github.com/ros2/rclcpp/issues/518
   Node::SharedPtr node;
-  try {
+  if (rclcpp::is_initialized()) {
     node = std::make_shared<Node>(std::forward<Args>(args) ...);
-  } catch (rclcpp::exceptions::RCLError e) {
-    // If node construction failed because ROS in not initialized,
-    // initialize ROS with no arguments and try again.
-    if (RCL_RET_NOT_INIT == e.ret) {
-      rclcpp::init(0, nullptr);
-      RCLCPP_INFO(internal_logger(),
-        "ROS was initialized without arguments.");
-      node = std::make_shared<Node>(std::forward<Args>(args) ...);
-    } else {
-      // For all other errors, pass exception up stack
-      RCLCPP_ERROR(internal_logger(),
-        "Failed to create node. Error: %s", e.what());
-      throw e;
-    }
+  } else {
+    rclcpp::init(0, nullptr);
+    RCLCPP_INFO(internal_logger(), "ROS was initialized without arguments.");
+    node = std::make_shared<Node>(std::forward<Args>(args) ...);
   }
 
   // Store shared pointer to static executor in this object

--- a/gazebo_ros/test/test_plugins.cpp
+++ b/gazebo_ros/test/test_plugins.cpp
@@ -43,6 +43,7 @@ protected:
 
 void TestPlugins::SetUp()
 {
+  std::cout << "Starting gzserver process with [" << GetParam().args[1] << "]" << std::endl;
   gazebo_process_ = std::make_unique<gazebo_ros::GazeboProcess>(GetParam().args);
   ASSERT_GT(gazebo_process_->Run(), 0);
 }

